core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -226,6 +259,135 @@ class HistoryServer(
 }
 
 /**
+ * An abstract implementation of the metrics [[Source]] trait with some common operations.
+ */
+private[history] abstract class HistoryMetricSource(val prefix: String) extends Source {
```

Comment:
Can you put this class (and the new gauges you're adding) in a new source file? They're not specific to `HistoryServer` so it's better to have them in a separate place.